datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -20,35 +24,45 @@
 
     private final BookModelAssembler assembler;
 
+
+
     public BookController(BookRepository repository, BookModelAssembler assembler) {
         this.repository = repository;
         this.assembler = assembler;
     }
 
-    @PostMapping("/books")
-    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book){
+    @PostMapping("/bookstores/{store_id}/books")
+    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book, @PathVariable Long store_id){
+        book.setStore_id(store_id);
         EntityModel<Book> entityModel = assembler.toModel(repository.save(book));
+
         return ResponseEntity
                 .created(entityModel.getRequiredLink(IanaLinkRelations.SELF).toUri())
                 .body(entityModel);
     }
     
-    @GetMapping("/books/{id}")
-    protected EntityModel<Book> one(@PathVariable Long id) {
+    @GetMapping("/bookstores/{store_id}/books/{id}")
+    protected EntityModel<Book> one(@PathVariable Long id, @PathVariable Long store_id) {
```

Comment:
What if the store id doesnt exist? Or what if the book id doesnt match the store id?